ic-os/components/setupos-scripts/devices.sh

```diff
@@ -27,17 +30,25 @@ function copy_config_files() {
     fi
 
     echo "* Copying SSH authorized keys..."
-    if [ -d "${CONFIG_DIR}/ssh_authorized_keys" ]; then
-        cp -r ${CONFIG_DIR}/ssh_authorized_keys /media/
-        log_and_halt_installation_on_error "${?}" "Unable to copy SSH authorized keys to hostOS config partition."
+    ssh_authorized_keys=$(get_config_value '.icos_settings.ssh_authorized_keys_path')
+    if [ -n "${ssh_authorized_keys}" ] && [ "${ssh_authorized_keys}" != "null" ]; then
+        if [ -d "${ssh_authorized_keys}" ]; then
+            cp -a "${ssh_authorized_keys}" /media/
+            log_and_halt_installation_on_error "${?}" "Unable to copy SSH authorized keys to hostOS config partition."
+        else
+            log_and_halt_installation_on_error "1" "Directory '${ssh_authorized_keys}' does not exist."
+        fi
     else
-        log_and_halt_installation_on_error "1" "Directory 'ssh_authorized_keys' does not exist."
+        echo "SSH authorized keys path is not configured."
```

Comment:
If this became a warning you may want to prefix text with `Warning: ` and also prefix `echo` with `>&2`

Reply:
Same everywhere else.

Reply:
Updated!